docs/RFCs/19042021_Cert_Management_In_CRDB_Helm_Chart.md

```diff
@@ -0,0 +1,243 @@
+  
+## Non-Goals
+ 
+* This RFC does not intend to fix the issues in the current default method of using Kubernetes CA
+ 
+## Helm Configuration
+This section specifies the suggested changes around user input in the Helm chart
+ 
+1. Add option specifying CockroachDB to manage the certificates, `tls.certs.generate.enabled` as true/false.  
+  Enabling this option will result in CockroachDB creating Node and Client certificates using a CA(either generated or provided by the user).
+ 
+2. Add option specifying CockroachDB to use user provided CA, `tls.certs.generate.caProvided` as true/false.  
+  Enabling this option will result in the generation of Node and Client certificates using the CA provided by the user.
+ 
+3. Add option specifying the secret name containing user provide CA, `tls.certs.generate.caSecret`.  
+  The secret name specified in this option will be used as a source for user-provided CA.
+  This option is mandatory if the `tls.certs.generate.caProvided` is true.
+ 
+4. Add option specifying the CA certificate expiration duration, `tls.certs.generate.caCertDuration`.  
+  This duration will only be used when we create our own CA. The duration value from this option will be used to set the expiry of the generated CA certificate.
+  By default, the CA expiry would be set to 10 years.
+ 
+5. Add option specifying the Client certificate expiration duration, `tls.certs.generate.clientCertDuration`.  
+  The duration value from this option will be used to set the expiry of the generated Client certificates. By default, Client certificate expiry would be set to 1 year.
+ 
+6. Add option specifying the Node certificate expiration duration, `tls.certs.generate.nodeCertDuration`.  
+  The duration value from this option will be used to set the expiry of the generated Node certificates. By default, Node certificate expiry would be set to 1 year.
+ 
+7. Add option specifying CockroachDB to manage rotation of the generated certificates, `tls.certs.generate.rotateCerts` as true/false.  
+  Enabling this option will result in auto-rotation of the certificates, before the expiry.
+ 
+## Helm Input Validation
+ 
+1. If `tls.certs.generate.caProvided` is set to true, then value for `tls.certs.generate.caSecret` must be provided.
+ 
+2. If value for `tls.certs.generate.caSecret` is provided, secret should exist in the CockroachDB install namespace.
+ 
+3. Value for `tls.certs.generate.caCertDuration` should be at least some delta months greater than the maximum of value for 
+   `tls.certs.generate.clientCertDuration` and `tls.certs.generate.nodeCertDuration`.
+ 
+ 
+## Implementation Details
+ 
+### Helm Components:
+ 
+When `tls.certs.generate.enabled` is set to `true`, the following components are created for certificate generation and rotation:
+ 1. Certificate Management Service as a `pre-install` job.
+ 2. ServiceAccount, for `pre-install` job. (deleted after pre-install hook succeeds)
```

Comment:
Does the service account need to stick around to support certificate rotation?

Reply:
The need of pre-install hook's service account is only for running the pre-install job to create certificates. This service account doesn't need to stick around because once the helm release has been successfully deployed, we can use crdb service account for running the cronjob later and use that to rotate certificates.

Reply:
+1, we should have a different SA for generating these certificates than for running CRDB nodes. It's good for the service account which runs CRDB nodes to have as few privileges as needed for it to run.

Reply:
Updated RFC doc to not to remove SA, Role and RoleBinding